core/src/avm2/activation.rs

```diff
@@ -2243,23 +2249,23 @@ impl<'a, 'gc> Activation<'a, 'gc> {
         let value2 = self.pop_stack().coerce_to_i32(self)?;
         let value1 = self.pop_stack().coerce_to_i32(self)?;
 
-        self.push_stack(value1.wrapping_mul(value2));
+        self.push_raw(value1.wrapping_mul(value2));
```

Comment:
I think this could also be applied to some more ops: `add`, `subtract`, `swap`, `astype`, `astypelate`? For the last three, since other ops guarantee that stack never has PrimitiveObject, then they can also never encounter a PrimitiveObject.